### ISSUE TITLE
bug in addoption with value with - characters in them

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#2687]
This issue was originally created at: 2010-09-03 09:22:49.
This issue was reported by: `jason_at_intel`.

>jason_at_intel said at 2010-09-03 09:22:49

There is a bug with option that habe '-' in them. Given this test case
```py
AddOption("--cfg-file","--config-file",
            dest='cfg_file',
            default='parts.cfg',
            nargs=1, type='string',
            action='store',
            help='Configuration file used to store common settings')
			
AddOption("--cfg",
            dest='build_config',
            default='debug',
            nargs=1, type='string',
            action='store',
            help='Configuration settings')
			
print(GetOption('build_config'))
print(GetOption('cfg_file'))
```

the command of

```
scons --cfg=foo
```
produces this:
```
scons: Reading SConscript files ...
debug
foo
...
```
It should have been 
```
foo
parts.cfg
```

gregnoel said at 2010-09-27 17:46:09
>Bug party triage.

>biziosan said at 2013-11-21 05:17:27

```py
    AddOption('--runtests',
              dest = 'runSuite',
              nargs = 1,
              type = 'string',
              action = 'store',
              default = None,
              help = 'Name of the set of tests to run.')
 
    AddOption('--runtest',
              dest = 'runTest',
              nargs = 1,
              type = 'string',
              action = 'store',
              default = None,
              help = 'Name of the test to run.')
 ```
produces the same bug. It seems to have something to do with the greediness of the regular expression that processes the command line.

>biziosan said at 2013-11-21 05:17:59

```py
    AddOption('--runtests',
              dest = 'runSuite',
              nargs = 1,
              type = 'string',
              action = 'store',
              default = None,
              help = 'Name of the set of tests to run.')
 
    AddOption('--runtest',
              dest = 'runTest',
              nargs = 1,
              type = 'string',
              action = 'store',
              default = None,
              help = 'Name of the test to run.')
```
produces the same bug. It seems to have something to do with the greediness of the regular expression that processes the command line.

> dirkbaechle said at 2014-05-17 08:08:33

Fixed for issue #2929, in revision https://bitbucket.org/scons/scons/commits/30f3640edaf6385e79899c917e5ffc00f97fc6bf

*** This issue has been marked as a duplicate of 2929 ***

>ipatrol said this issue depends on #2763 at 2011-05-14 12:29:10.

>dirkbaechle said this issue  is a duplicate of #2929 at 2014-05-17 08:08:33.
